components/ProfileUploadPhoto.vue

```diff
@@ -0,0 +1,180 @@
+<script setup lang="ts">
+import { MAX_FILE_SIZE_MB } from "@/utils/constants/maxFileSizeMB";
+
+const {
+  isDragActive,
+  errorMessages,
+  root,
+  handleDrag,
+  handleDrop,
+  handleKeydown,
+  resetState,
+  handleUpload,
+  uploadedFiles,
+} = useDragAndDrop(MAX_FILE_SIZE_MB, 1);
+
+const { upsertProfileImage, checkProfileImage, profileImageError } =
+  useProfileImage();
+
+const profileImageSource = useProfileImageSource();
+
+watch(
+  uploadedFiles,
+  () => {
+    upsertProfileImage(uploadedFiles.value[0]);
+  },
+  { deep: true }
+);
+
+onMounted(() => {
+  checkProfileImage();
+});
+</script>
+
+<template>
+  <section ref="root" class="profile__photo">
+    <span class="profile__photo--name">Photo</span>
+    <div
+      class="profile__photo--dropzone"
+      :class="{ 'profile__photo__dropzone--active': isDragActive }"
+      @drop.prevent="handleDrop"
+      @dragenter="handleDrag"
+      @dragover.prevent="handleDrag"
+      @dragleave="handleDrag"
+    >
+      <p class="profile__photo__description">
+        Drop your photo here <br />
+        or
+      </p>
+      <img
+        class="profile__photo--image"
+        :src="profileImageSource"
+        width="400"
+        height="400"
+      />
+    </div>
+    <label
+      class="profile__photo__label"
+      for="file"
+      tabindex="0"
+      role="button"
+      aria-pressed="false"
+      @keydown="handleKeydown"
+      @click="resetState"
+    >
+      Upload your photo
+      <img
+        class="profile__photo__icon"
+        src="@/assets/img/upload.png"
+        alt="upload"
+      />
+      <input
+        id="file"
+        class="profile__photo__input"
+        type="file"
+        name="file"
+        multiple
+        @change="handleUpload"
+      />
+    </label>
+    <template v-if="errorMessages.length">
+      <BaseNotification
+        v-for="error in errorMessages"
+        :key="error"
+        theme="error"
+        >{{ error }}</BaseNotification
+      >
+    </template>
+    <BaseNotification v-if="profileImageError" theme="error">{{
```

Comment:
Completed, there is useNotification composable that contains the logic for BaseNotification. The BaseNotification is added to app.vue. Additionally there is usePreventBodyScroll composable used in useNotification to prevent body scrolling when notification is open.